src/data/routes.csv

```diff
@@ -1,109 +1,109 @@
-71,5.11c,6c+,8-,23,9a,7
-72,5.11c,7a,8,23,9b,7+
-73,5.11d,7a,8,23,9b,7+
-74,5.11d,7a+,8+,24,9c,8-
-75,5.12a,7a+,8+,24,9c,8-
-76,5.12b,7b,8+/9-,25,10a,8-
-77,5.12b,7b,8+/9-,25,10a,8-
-78,5.12c,7b+,9-,26,10a,8
-79,5.12c,7b+,9-,26,10a,8
-80,5.12d,7c,9,27,10b,8+
-81,5.12d,7c,9,27,10b,8+
-82,5.13a,7c+,9+,28,10c,8+
-83,5.13a,7c+,9+,28,10c,8+
-84,5.13b,8a,10-,29,10c,9-
-85,5.13b,8a,10-,29,10c,9-
-86,5.13c,8a+,10,30,11b,9-
-87,5.13c,8a+,10,30,11b,9-
-88,5.13d,8b,10+,31,11c,9
-89,5.13d,8b,10+,31,11c,9
-90,5.14a,8b+,10+/11-,32,11c,9+
-91,5.14a,8b+,10+/11-,32,11c,9+
-92,5.14b,8c,11-,33,12a,9+
-93,5.14b,8c,11-,33,12a,9+
-94,5.14c,8c+,11,34,12b,10-
-95,5.14c,8c+,11,34,12b,10-
-96,5.14d,9a,11,35,12c,10
-97,5.14d,9a,11,35,12c,10
-98,5.15a,9a+,11/11+,36,13a,10+
-99,5.15a,9a+,11/11+,36,13a,10+
-100,5.15b,9b,11+,37,13b,10+
-101,5.15b,9b,11+,37,13b,10+
-102,5.15c,9b+,11+/12-,38,13b,11-
-103,5.15c,9b+,11+/12-,38,13b,11-
-104,5.15d,9c,12-,39,13c,11
-105,5.15d,9c,12-,39,13c,11
-106,5.16a,9c+,12,40,13c,11+
-107,5.16a,9c+,12,40,13c,11+
+score,yds,french,uiaa,ewbank,saxon,norwegian,brazilian
+0,5.0,1a,1,1,1,1-,I
+1,5.0,1a,1,1,1,1-,I
+2,5.0,1a+,1,1,1,1-,I
+3,5.0,1a+,1,1,1,1-,I
+4,5.0,1b,1,1,1,1,I
+5,5.0,1b,1,2,1,1,I
+6,5.0,1b+,1,2,1,1+,I
+7,5.0,1b+,1,2,1,1+,I
+8,5.0,1c,1,2,1,1+,Isup
+9,5.0,1c,1,2,1,1+,Isup
+10,5.0,1c+,1,3,1,2-,Isup
+11,5,1c+,1,4,1,2-,Isup
+12,5.1,2a,2,5,2,2,Isup
+13,5.1,2a,2,5,2,2,Isup
+14,5.1,2a+,2,5,2,2,II
+15,5.1,2a+,2,6,2,2,II
+16,5.1,2b,2,6,2,2+,II
+17,5.1,2b,2,6,2,2+,II
+18,5.2,2b+,2,7,2,2+,II
+19,5.2,2b+,2,7,2,2+,II
+20,5.2,2c,2,7,2,3-,II
+21,5.2,2c,2,7,2,3-,II
+22,5.3,2c+,2,8,2,3,IIsup
+23,5.3,2c+,2,8,2,3,IIsup
+24,5.3,3a,3,9,3,3,IIsup
+25,5.3,3a,3,9,3,3,IIsup
+26,5.4,3a+,3,10,3,3+,IIsup
+27,5.4,3a+,3,10,3,3+,IIsup
+28,5.4,3b,3,10,3,3+,III
+29,5.4,3b,3,10,3,3+,III
+30,5.4,3b+,3,10,3,4-,III
+31,5.5,3b+,3,11,3,4-,III
+32,5.5,3c,3,11,3,4,III
+33,5.5,3c,3,12,3,4,III
+34,5.5,3c+,3,12,3,4,III
+35,5.5,3c+,3,12,3,4,III
+36,5.6,4a,4,13,4,4+,IIIsup
+37,5.6,4a,4,13,4,4+,IIIsup
+38,5.6,4a+,4+,13,4,4+,IIIsup
+39,5.6,4a+,4+,13,4,4+,IIIsup
+40,5.6,4b,4+,13,5,5-,IIIsup
+41,5.7,4b,4+,14,5,5-,IIIsup
+42,5.7,4b+,5-,14,5,5-,IV
+43,5.7,4b+,5-,14,5,5-,IV
+44,5.7,4c,5-,15,5,5,IV
+45,5.7,4c,5-,15,5,5,IV
+46,5.8,4c+,5,16,6,5,IV
+47,5.8,4c+,5,16,6,5,IV
+48,5.8,5a,5,16,7a,5+,IVsup
+49,5.8,5a,5,16,7a,5+,IVsup
+50,5.8,5a+,5+,16,7a,5+,IVsup
+51,5.8,5a+,5+,16,7a,5+,IVsup
+52,5.9,5b,5+,17,7a,5+,IVsup
+53,5.9,5b,5+,17,7a,5+,IVsup
+54,5.9,5b+,6-,17,7b,5+,IVsup
+55,5.9,5b+,6-,17,7b,5+,IVsup
+56,5.9,5c,6,17,7b,6-,V
+57,5.10a,5c,6,18,7b,6-,V
+58,5.10a,5c+,6+,18,7c,6-,V
+59,5.10a,5c+,6+,18,7c,6-,V
+60,5.10a,6a,6+,18,7c,6-,V
+61,5.10a,6a,6+,18,7c,6-,V
+62,5.10b,6a+,7-,19,8a,6,Vsup
+63,5.10b,6a+,7-,19,8a,6,Vsup
+64,5.10c,6b,7,20,8b,6,VI
+65,5.10c,6b,7,20,8b,6,VI
+66,5.10d,6b+,7+,20,8c,6+,VIsup
+67,5.11a,6b+,7+,21,8c,6+,VIsup
+68,5.11a,6c,7+/8-,21,9a,7-,VIIa
+69,5.11b,6c,7+/8-,22,9a,7-,VIIa
+70,5.11b,6c+,8-,22,9a,7,VIIb
+71,5.11c,6c+,8-,23,9a,7,VIIb
+72,5.11c,7a,8,23,9b,7+,VIIc

```

Comment:
Yeah, to make it as brief as possible: the brazilian grade standard actually has two grades for any route that IS NOT a single pitch/sports route:

- the general grade of the route
- the grade of the hardest move (crux)

for the general grade, an arabic number is used: i.e. 6
for the crux grade, it's roman. so a route that is generally graded in 5th grade but has a 6sup grade crux, it reads like:
`5º 6sup`

for sports climb/single pitch, we use only the second form because there's no point in having two assignments.
I have focused my implementation in just the second form. However, later we could expand to enable the whole aspect of the brazilian specification, which has more to it: duration, artificial grade, exposition grade.

Reply:
Ok, this clarifies things!  

> for the general grade, an arabic number is used: i.e. 6
> for the crux grade, it's roman. so a route that is generally graded in 5th grade but has a 6sup grade crux, it reads like:
> `5º 6sup`

Should that be `5º VIsup`?

Reply:
Definitely 😨

Reply:
>  I have focused my implementation in just the second form. However, later we could expand to enable the whole aspect of the brazilian specification, which has more to it: duration, aid grade, exposition grade.

Yes, makes sense to start with the common case.
 
We will need a system to handle this for other scales too. I haven't worked out the details, but I plan to allow hybrid/combined grades that combine complementary scales.  For example, a British trad grade could be constructed from a grade on each of the adjectival and technical scales, e.g. `BritishAdjectival("E6")` + `BritishTechnical("6a")`. Likewise, an aid route with mandatory climbing could get a grade of `5.8 A1 `, which would be parsed as `Aid("A1")` + `YosemiteDecimal("5.5")`.  

If I understand the spec you shared, most Brazilian grades would be represented by a combination of

- `BrazilianGeneral` ~ `[1º, 2º, 3º, 4º, 5º, 6º, 7º, 8º,…]`
- `BrazilianTechnical` ~ `['I', 'Isup', 'II', 'IIsup',...]` <- this PR
- `BrazilianAid` == `Aid`?
- `BrazilianDuration` ~ ['D1', 'D2', ..., 'D7']
- `BrazilianExposure` ~  ['E1', 'E2', ..., 'E5']


Does this general idea make sense for Brazilian grades?

Reply:
Yes, I believe that's one way to approach it!

Reply:
I'll let you suggest a name for the scale in this PR

Reply:
As per the spec, this is the grade of the hardest move on the route (or the crux move). I'd go with `BrazilianCrux`, which may seem weird at first but once you get the picture it sort of makes sense. What do you think?

Reply:
ok!